packages/block-library/src/navigation-link/shared/controls.js

```diff
@@ -149,18 +162,20 @@ export function Controls( { attributes, setAttributes, clientId } ) {
 					__next40pxDefaultSize
 					id={ inputId }
 					label={ __( 'Link' ) }
-					value={ url ? safeDecodeURI( url ) : '' }
-					onChange={ ( urlValue ) => {
-						if ( hasUrlBinding ) {
-							return; // Prevent editing when URL is bound
-						}
-						setAttributes( {
-							url: encodeURI( safeDecodeURI( urlValue ) ),
-						} );
-					} }
+					value={ inputValue ? safeDecodeURI( inputValue ) : '' }
 					autoComplete="off"
 					type="url"
 					disabled={ hasUrlBinding }
+					onChange={ ( newValue ) => {
+						if ( hasUrlBinding ) {
+							return;
+						}
+
+						// Don't update the actual url attribute yet, defer to onBlur so we
+						// don't end up with the canvas thinking the url is empty, which causes
+						// the label to be replaced by the placeholder text.
```

Comment:
```suggestion
						// Defer updating the url attribute until onBlur to prevent the canvas from 
						// treating it as empty and replacing the label with placeholder text.
```

Nit